controllers/kubeadmconfig_controller.go

```diff
@@ -400,6 +407,42 @@ func (r *KubeadmConfigReconciler) createClusterCertificates(ctx context.Context,
 	return certificates, nil
 }
 
+func (r *KubeadmConfigReconciler) createKubeconfigSecret(ctx context.Context, clusterName, endpoint, namespace string, certificates *certs.Certificates) error {
+	cert, err := certs.DecodeCertPEM(certificates.ClusterCA.Cert)
+	if err != nil {
+		return errors.Wrap(err, "failed to decode CA Cert")
+	} else if cert == nil {
+		return errors.New("certificate not found in config")
+	}
+
+	key, err := certs.DecodePrivateKeyPEM(certificates.ClusterCA.Key)
+	if err != nil {
+		return errors.Wrap(err, "failed to decode private key")
+	} else if key == nil {
+		return errors.New("CA private key not found")
+	}
+
+	server := fmt.Sprintf("https://%s", endpoint)
+	cfg, err := certs.NewKubeconfig(clusterName, server, cert, key)
+	if err != nil {
+		return errors.Wrap(err, "failed to generate a kubeconfig")
+	}
+
+	yaml, err := clientcmd.Write(*cfg)
+	if err != nil {
+		return errors.Wrap(err, "failed to serialize config to yaml")
+	}
+
+	secret := &corev1.Secret{}
+	secretName := fmt.Sprintf("%s-kubeconfig", clusterName)
```

Comment:
We should probably add a method to `sigs.k8s.io/cluster-api/util` for generating this name, since we're going to need it in other areas of cluster-api outside of the bootstrap provider as well. Not a blocker for this PR, but we should at least create an issue to track it.

Reply:
Ah cool. Want me to make an issue and do it? Im game.

Reply:
Please do, thanks!

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/1266